### PR TITLE
Change AWS_EXECUTION_ENV set in the lambda init binary

### PR DESCRIPTION
**Pull request by dfangl**
## Motivation
The lambda init binary should have the `AWS_EXECUTION_ENV` set to `AWS_Lambda_rapid`. The setting of the concrete value depending on the runtime will be done in `/var/runtime/bootstrap`, and is part of the lambda images.

Some extensions depend on that value being correct for checking what runtime is running (datadog).

Even though this is a parity adjustment, it is a quite significant change, so I labeled it minor for now.

## Changes
* Change execution env variable to `AWS_Lambda_rapid`
* Allow a custom init even if `LAMBDA_INIT_DEBUG` is not set
* Add test for introspection of the init process environment

### CODE REVIEW

File: localstack/services/lambda_/invocation/docker_runtime_executor.py
Comment (joe4dev) [reply localstack/services/lambda_/invocation/docker_runtime_executor.py:361]: (already hit me 😅 when trying to test the changes while having `LAMBDA_INIT_BIN_PATH` set by default)

File: tests/aws/services/lambda_/test_lambda.py
Comment (joe4dev) [inline tests/aws/services/lambda_/test_lambda.py:701]: Should we group them similar to `tests.aws.services.lambda_.test_lambda_common.TestLambdaRuntimesCommon.test_introspection_invoke`?
It would be helpful to see what's actionable and what's simply LocalStack extras. For some, I guess we don't care.

File: tests/aws/services/lambda_/test_lambda.py
Comment (joe4dev) [inline tests/aws/services/lambda_/test_lambda.py:700]: I'm getting this diff when validating against AWS:

```
>> match key: create-result
	(+) /CreateFunctionResponse/LoggingConfig ( {'LogFormat': 'Text', 'LogGroup': '/aws/lambda/<function-name:1>'} )
```

(Initially wanted to double-check `"_AWS_XRAY_DAEMON_ADDRESS": "169.254.79.129",` but then noticed this one; might be a very recent update)

**Review by joe4dev — APPROVED** [review]
Love the new init environment test 🤩 

I don't expect people to customize their init environment for LocalStack based on the runtime suffix we had for `AWS_EXECUTION_ENV`.

Nothing blocking the merge; just added some comments/suggestions (e.g., regarding our internal dev flow).

**Comment by joe4dev** [comment]
Obviously, `LOCALSTACK_POST_INVOKE_WAIT_MS` needs to be ignored as well to unblock the CI:

```
>> match key: lambda-init-inspection
	#x1B[32m(+)#x1B[0m /Payload/environment/LOCALSTACK_POST_INVOKE_WAIT_MS ( '50' )
```

File: localstack/services/lambda_/invocation/execution_environment.py
Comment (dfangl) [reply localstack/services/lambda_/invocation/execution_environment.py:157]: I think that it may not be propagated to the child processes of the init, as the runtime sets it itself, and the extensions on AWS do not have it set at all. So I think the comment is not valid here anymore, missed it!

File: localstack/services/lambda_/invocation/docker_runtime_executor.py
Comment (dfangl) [reply localstack/services/lambda_/invocation/docker_runtime_executor.py:361]: The lambda init path was necessary anyway before. Now you can set only the init path to enable usage of a new lambda init binary without having to setup debugging itself.
You cannot toggle the init copying with the debug variable, but I would suggest using different profiles for this, instead of the toggle of init debug?

I can leave it out from this PR if you do not think it improves usability for developers, but for me it simplifies it, because I more often try out new inits without debugging than with.

File: tests/aws/services/lambda_/test_lambda.py
Comment (dfangl) [reply tests/aws/services/lambda_/test_lambda.py:751]: No particular reason, copied the whole thing from another test to be honest.

File: tests/aws/services/lambda_/test_lambda.py
Comment (dfangl) [reply tests/aws/services/lambda_/test_lambda.py:753]: Same, was copied, will remove it and let it be default.

File: localstack/services/lambda_/invocation/execution_environment.py
Comment (dominikschubert) [inline localstack/services/lambda_/invocation/execution_environment.py:159]: great catch :+1:

**Review by dominikschubert — APPROVED** [review]
LGTM, I don't have a strong opinion in either direction regarding the init flag changes though :+1:

File: localstack/services/lambda_/invocation/docker_runtime_executor.py
Comment (joe4dev) [reply localstack/services/lambda_/invocation/docker_runtime_executor.py:361]: I agree that we should have a way to simplify init testing without debugging 👍 

Using profiles makes a lot of sense here 💯 . Then having two `entrypoint.sh` versions, one for testing (`exec /var/rapid/init`) and one for debugging makes switching easier. Thanks for sharing 👍

**Pull request merged**
